tools/c7n_azure/readme.md

```diff
@@ -1,49 +1,64 @@
-Service Principal Authentication
---------------------------------
 
-To authenticate via environment variables, you can create a service principal with Azure CLI
+# Cloud Custodian - Azure Support
 
-```
-# select correct subscription
-az account set -s "my subscription name"
+This a plugin to Cloud Custodian that adds Azure support.
 
-# create service principal
-az ad sp create-for-rbac --name <name> --password <password>
-```
-This will yield something like:
+## Install Cloud Custodian and Azure Plugin
 
-```
-{
-  "appId": appid,
-  "displayName": name,
-  "name": name,
-  "password": password,
-  "tenant": guid
-}
-```
-You will need to map it to environment variables like this:
+The Azure provider must be installed as a separate package in addition to c7n.
 
-AZURE_TENANT_ID=tenant
-AZURE_SUBSCRIPTION_ID=subscription id
-AZURE_CLIENT_ID=appId
-AZURE_CLIENT_SECRET=password
+    $ virtualenv custodian
+    $ source custodian/bin/activate
+    (custodian) $ pip install c7n
+    (custodian) $ pip install c7n_azure
 
 
+## Write your first policy
 
-Azure CLI Authentication
-------------------------
+A policy specifies the following items:
 
-Set environment variable `AZURE_CLI_AUTH` to any value, and session will pull credentials from cached AZ Login and
-default subscription.
+- The type of resource to run the policy against
+- Filters to narrow down the set of resources
+- Actions to take on the filtered set of resources
 
+For this tutorial we will add a tag to all virtual machines with the name "Hello" and the value "World".
+
+Create a file named ``custodian.yml`` with this content:
+
+    policies:
+        - name: my-first-policy
+          description: |
+            Adds a tag to all virtual machines
+          resource: azure.vm
+          actions:
+            - type: tag
+              tag: Hello
```

Comment:
looks good, and this is just docs, but fwiw, its also nice to support tagging multiple values in a single action impl

Reply:
Oh yeah that action also accepts `tags: { dictionary }`